test/integration/money.exs

```diff
@@ -0,0 +1,34 @@
+defmodule Gringotts.Integration.Gateways.MoneyTest do
+  use ExUnit.Case, async: true
+
+  alias Gringotts.Money, as: MoneyProtocol
+
+  @moduletag :integration
+  
+  @ex_money Money.new(42, :EUR)
+  @ex_money_long Money.new("42.126456", :EUR)
+  @ex_money_bhd Money.new(42, :BHD)
+  
```

Comment:
We should add test cases to cover `Any` implementation as well. test cases would help understand our implementation better.

Reply:
this test suit has to be more extensive IMO.